pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java

```diff
@@ -258,32 +257,26 @@ public static void convertFromLegacyTableConfig(TableConfig tableConfig) {
     IndexingConfig indexingConfig = tableConfig.getIndexingConfig();
 
     if (streamIngestionConfig == null) {
-      Map<String, String> streamConfigs = indexingConfig.getStreamConfigs();
-
       // Only set the new config if the deprecated one is set.
-      if (streamConfigs != null && !streamConfigs.isEmpty()) {
+      Map<String, String> streamConfigs = indexingConfig.getStreamConfigs();
+      if (MapUtils.isNotEmpty(streamConfigs)) {
         streamIngestionConfig = new StreamIngestionConfig(Collections.singletonList(streamConfigs));
       }
     }
 
     if (ingestionConfig == null) {
       if (batchIngestionConfig != null || streamIngestionConfig != null) {
-        ingestionConfig = new IngestionConfig(batchIngestionConfig, streamIngestionConfig, null, null, null, null);
-      }
-    } else {
-      if (batchIngestionConfig != null) {
+        ingestionConfig = new IngestionConfig();
```

Comment:
shall we always create an empty IngestionConfig ? Does null represent something else?

Reply:
Currently empty ingestion config has the same behavior with `null`, but usually we expect `null` when the config does not apply